public/input_text_load_data.js

```diff
@@ -0,0 +1,104 @@
+
+/*takes a string of data returns an array of that data*/
+function parseData(data, collection) {
+  var split = data.split('--ML_BOUNDARY');
+  var items = [];
+
+  for (var i=0; i < split.length; i++) {
+    var item = {
+      category: null,
+      content: null,
+      contentLength: null,
+      contentType: null,
+      format: null,
+      uri: null
+    };
+
+    var matches = split[i].match(/Content-Type: ([\w\/]+)/);
+    if(matches && matches[1]) {
+      item.contentType = matches[1];
+    }
+
+    var matches2 = split[i].match(/Content-Disposition: ([\w\/]+); filename="([^"]+)"; category=([\w\/]+); format=([\w\/]+)/);
+    var matches3 = split[i].match(/Content-Length: ([\d]+)/);
+    var matches4 = split[i].match(/({[^$]*})/);
+
+    if(matches2 && matches2[2]) {
+      item.uri = matches2[2];
+      if(matches2[3]) {
+        item.category = matches2[3];
+      }
+      if(matches2[4]) {
+        item.format = matches2[4];
+      }
+    }
+    if(matches3 && matches3[1]) {
+      item.contentLength = matches3[1];
+    }  
+    if(matches4 && matches4[1]) {
+      item.content = JSON.parse(matches4[1]);
+    }
+
+    if(item.content) {
+      if (collection && ((collection.indexOf('.') !== -1 && item.uri.substring(0, collection.indexOf('.')) === collection.substring(0, collection.indexOf('.'))) || collection === item.uri)) {
```

Comment:
Before adding an object to the array, it checks to see that 
1.) the physical document has some content/data (not null)
and either 
2.) the collection exists and matches the file name of the physical doc
or
3.) the collection exists and the substring up to the '.' matches the substring of the physical doc's uri up to the doc. So addr.583495733492.json would match addr.json, since addr matches addr. 

If the condition is met, then push to the array.